File: src/dataset/go/collatz_conjecture_Exercism_go_ThomasZumsteg/gemini/GeminiFlash_collatz_conjecture_v2.go
package collatzconjecture

import "fmt"

func CollatzConjecture(num int) (int, error) {
	if num <= 0 {
		return 0, fmt.Errorf("Integers must be greater than zero: %d", num)
	}

	steps := 0
	for num != 1 {
		if num&1 == 0 { // Use bitwise AND for even number check, faster than modulo
			num >>= 1 // Use right bit shift for division by 2, faster than division
		} else {
			num = num*3 + 1
		}
		steps++
	}
	return steps, nil
}

File: src/dataset/c/collatz-conjecture_exercism-Manoloon/gemini/GeminiFlash_collatz_conjecture_v2.c
#include "collatz_conjecture.h"

int steps(int start) {
  if (start <= 0) {
    return ERROR_VALUE;
  }

  if (start == 1) {
    return 0;
  }

  int step_count = 0;
  while (start > 1) {
    if ((start & 1) == 0) { // Check if even using bitwise AND
      start >>= 1;          // Divide by 2 using right bit shift
    } else {
      if (start > (INT_MAX - 1) / 3) return ERROR_VALUE; // Check for potential overflow before multiplication
      start = (start * 3) + 1;
    }
    step_count++;
  }
  return step_count;
}